src/Castle.Core.AsyncInterceptor/AsyncInterceptorBase.cs

```diff
@@ -95,7 +95,14 @@ private static void InterceptSynchronousVoid(AsyncInterceptorBase me, IInvocatio
             // If the intercept task has yet to complete, wait for it.
             if (!task.IsCompleted)
             {
-                Task.Run(() => task).Wait();
+                try
+                {
+                    Task.Run(() => task).Wait();
+                }
+                catch (AggregateException ex)
+                {
+                    throw ex.InnerException;
```

Comment:
Note this will drop the exception stack trace. I’d recommend just reusing some copy-pasta code from SO for rethrowing exceptions to make sure the user gets the trace they expected.

Did this end up giving to the user double nested AggregateExceptions though? From an async method AggregateExceptions might actually be expected

Reply:
I'm still figuring it out TBH.

One of the tests were failing because `AggregateException` was being thrown not the expected exception, this did fix the issue.

I'll check to see if it looses the stack trace.

Reply:
Whoops, sorry for jumping on it too soon!

Reply:
No worries, it's valued feedback, and I hadn't considered the stack trace, or more accurately I had assumed it would be retained, but I hadn't tested it.

Reply:
Perhaps this?

```csharp
ExceptionDispatchInfo.Capture(ex.InnerException).Throw();
```